gcloud/storage/bucket.py

```diff
@@ -124,8 +124,14 @@ def exists(self):
             # We only need the status code (200 or not) so we seek to
             # minimize the returned payload.
             query_params = {'fields': 'name'}
+            # We intentionally pass `_target_object=None` since fields=name
+            # would limit the local properties.
             self.connection.api_request(method='GET', path=self.path,
-                                        query_params=query_params)
+                                        query_params=query_params,
+                                        _target_object=None)
+            # NOTE: This will not fail immediately in a batch. However, when
+            #       Batch.finish() is called, the resulting `NotFound` will be
+            #       raised.
```

Comment:
@tseaver comment from f48620fd5a4d2014eba0772a88698dd2642d60fb

> Ugh, that is going to be nasty for app developers to figure out.

----

Yeah I just stopped short of having the comment be an `XXX` to mark it was very concerning.

I'm not sure what we want to do here. In some sense it gives the "right" behavior "eventually" but instead of

```python
>>> val = blob.exists()
>>> val
False
```

we get

```python
>>> val = None
>>> with Batch():
...     val = blob.exists()
... 
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
NotFound
>>> val
True
```